compiler/rustc_mir/src/transform/large_enums.rs

```diff
@@ -0,0 +1,251 @@
+                            }
+                            let alloc = interpret::Allocation::from_bytes(
+                                data,
+                                tcx.data_layout.ptr_sized_integer().align(&tcx.data_layout).abi,
+                                Mutability::Not,
+                            );
+                            let alloc = tcx.intern_const_alloc(alloc);
+                            alloc_cache.insert(ty, alloc);
+                            // FIXME(jknodt) use entry API
+                            alloc_cache.get(ty).unwrap()
+                        };
+
+                        let tmp_ty = tcx.mk_ty(ty::Array(
+                            tcx.types.usize,
+                            Const::from_usize(tcx, num_variants),
+                        ));
+
+                        let size_array_local = patch.new_temp(tmp_ty, span);
+                        let store_live = Statement {
+                            source_info,
+                            kind: StatementKind::StorageLive(size_array_local),
+                        };
+
+                        let place = Place::from(size_array_local);
+                        let constant_vals = Constant {
+                            span,
+                            user_ty: None,
+                            literal: ConstantKind::Val(
+                                interpret::ConstValue::ByRef { alloc, offset: Size::ZERO },
+                                tmp_ty,
+                            ),
+                        };
+                        let rval = Rvalue::Use(Operand::Constant(box (constant_vals)));
+
+                        let const_assign = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (place, rval)),
+                        };
+
+                        let discr_place =
+                            Place::from(patch.new_temp(adt_def.repr.discr_type().to_ty(tcx), span));
+
+                        let store_discr = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                discr_place,
+                                Rvalue::Discriminant(*rhs),
+                            )),
+                        };
+
+                        // FIXME(jknodt) do I need to add a storage live here for this place?
+                        let size_place = Place::from(patch.new_temp(tcx.types.usize, span));
+
+                        let store_size = Statement {
+                            source_info,
+                            kind: StatementKind::Assign(box (
+                                size_place,
+                                Rvalue::Use(Operand::Copy(Place {
+                                    local: size_array_local,
+                                    projection: tcx
+                                        .intern_place_elems(&[PlaceElem::Index(discr_place.local)]),
```

Comment:
We still need to cast `discr_place` to an `usize` before using it in indexing projection. If this is responsible for CI failure, it would be nice to extend MIR validator with extra check to catch that in a separate PR.

Reply:
will get to this soon, sorry for the delay